src/ui/i18n/i18n.js

```diff
@@ -0,0 +1,107 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let registeredTranslations = {};
```

Comment:
Moving the default locale out of the module scope was good, and the implementation makes things even easier to follow in my eyes, but this value is also "module-level" state that should be moved into per-server state somewhere.

What if we modified these methods to always accept a list of files?

- `getRegisteredTranslationLocales` => `getTranslationLocales(translationPaths)`
  - return the `getLocaleFromFileName()` for each path
- `getTranslations(... languageTags)` => `getTranslations(translationPaths, ... languageTags)`
- remove `registerTranslations()`, `unregisterTranslations()`, and `getRegisteredTranslations()`

If we go this route then we will need to extend the `UiI18n` class again to collect the language files exported by each plugin and we'll definitely want to start unit testing that class. The `_build:verifyTranslations` task should be really easy to port over, since it's already dealing with a list of paths that it can start passing in.